modules/ipc/src/zenoh/service.cpp

```diff
@@ -2,4 +2,79 @@
 // Copyright (C) 2023-2024 HEPHAESTUS Contributors
 //=================================================================================================
 
-#include "hephaestus/ipc/zenoh/service.h"  // NOLINT(misc-include-cleaner)
+#include "hephaestus/ipc/zenoh/service.h"
+
+#include <chrono>
+#include <cstddef>
+#include <cstdint>
+#include <span>
+#include <string>
+#include <utility>
+#include <variant>
+#include <vector>
+
+#include <fmt/format.h>
+#include <zenoh.h>
+#include <zenoh/api/base.hxx>
+#include <zenoh/api/channels.hxx>
+#include <zenoh/api/encoding.hxx>
+#include <zenoh/api/keyexpr.hxx>
+#include <zenoh/api/reply.hxx>
+#include <zenoh/api/session.hxx>
+
+#include "hephaestus/ipc/topic.h"
+#include "hephaestus/ipc/zenoh/conversions.h"
+#include "hephaestus/ipc/zenoh/session.h"
+#include "hephaestus/telemetry/log.h"
+
+namespace heph::ipc::zenoh {
+namespace {
+[[nodiscard]] auto
+getServiceCallResponses(const ::zenoh::channels::FifoChannel::HandlerType<::zenoh::Reply>& service_replies)
+    -> std::vector<ServiceResponse<std::vector<std::byte>>> {
+  std::vector<ServiceResponse<std::vector<std::byte>>> reply_messages;
+  for (auto res = service_replies.recv(); std::holds_alternative<::zenoh::Reply>(res);
+       res = service_replies.recv()) {
```

Comment:
to me its not clear, why you have to receive twice here, an explanatory comment would be great

Reply:
this is a for loop, the first recv is the initialization, the second is the increment